internal/provider/provider.go

```diff
@@ -150,6 +151,28 @@ func New() *schema.Provider {
 					},
 				},
 			},
+			"coder_host": {
+				Description: "Use this data source to get information about the host.",
+				ReadContext: func(c context.Context, rd *schema.ResourceData, i interface{}) diag.Diagnostics {
+					rd.SetId(uuid.NewString())
+					rd.Set("os", runtime.GOOS)
+					rd.Set("arch", runtime.GOARCH)
+
+					return nil
+				},
+				Schema: map[string]*schema.Schema{
+					"os": {
```

Comment:
nit: Is it worth limiting the values to https://github.com/golang/go/blob/master/src/go/build/syslist.go or linking to the known values in the docs so users don't have to do guesswork? 

🤷🏼 I'm not sure if anyone would use this beyond what we're doing in the Docker template though.

Reply:
I'll mention that it's the same as `GOOS`, and link to https://pkg.go.dev/runtime#pkg-constants.